content/solids/components/voxel-painter.ts

```diff
@@ -0,0 +1,462 @@
+      $button.on('click', () => {
+        let outputString = '"'
+        for (let i = 0; i < voxels.length; i++)
+          for (let j = 0; j < 3; j++)
+            outputString += voxels[i].position.getComponent(j) + ","
+        outputString += '"'
+        prompt('In order to load the current voxels in, copy the below, then paste it next to shape= ', outputString)
+      })
+    }
+
+    {
+      const ambientLight = new THREE.AmbientLight(0x606060);
+      scene.add(ambientLight);
+
+      const directionalLight = new THREE.DirectionalLight(0xffffff);
+      directionalLight.position.set(1, 0.75, 0.5).normalize();
+      scene.add(directionalLight);
+    }
+
+    let rendererSize = renderer.getSize()
+    // const camera = new THREE.PerspectiveCamera(45, rendererSize.width / rendererSize.height, 1., 10000.);
+    const cameraW = 30.
+    const cameraH = cameraW * rendererSize.height /rendererSize.width
+    const camera = new THREE.OrthographicCamera(-cameraW / 2., cameraW / 2., cameraH / 2., -cameraH / 2., 1., 10000.)
+    camera.rotation.order = "YXZ"
+    camera.rotation.y = TAU/8.
+    camera.rotation.x =-TAU / 8.
+    scene.add(camera)
+
+    const voxelGeo = new THREE.BoxGeometry(1, 1, 1);
+    const voxelMaterial = new THREE.MeshLambertMaterial({color: 0xfeb74c});
+    if ( this.attr('colorSides') === "true")
+    {
+      voxelGeo.faces[ 0].color = new THREE.Color(1.,.5,.5)
+      voxelGeo.faces[ 2].color = new THREE.Color(0.,1.,0.)
+      voxelGeo.faces[ 4].color = new THREE.Color(1.,1.,0.)
+      voxelGeo.faces[ 6].color = new THREE.Color(1.,0.,1.)
+      voxelGeo.faces[ 8].color = new THREE.Color(0.,1.,1.)
+      voxelGeo.faces[10].color = new THREE.Color(1., 1., 1.)
+      for(let i = 0; i < 6; ++i)
+        voxelGeo.faces[i * 2 + 1].color = voxelGeo.faces[i * 2 + 0].color
+
+      voxelMaterial.vertexColors = THREE.FaceColors
+      voxelMaterial.color.setRGB(1.,1.,1.)
+    }
+      
+    let outlineGeometry = new THREE.Geometry()
+    let outlineMaterial = new THREE.LineBasicMaterial()
+    let cubeEdges = [0,1,2,3,5,4,7,6, 0,2,1,3,5,7,4,6, 0,5,1,4,2,7,3,6]
+    for (let i = 0; i < cubeEdges.length; i++)
+      outlineGeometry.vertices[i] = voxelGeo.vertices[cubeEdges[i]].clone()
+
+    const placementVisualizer = new THREE.Mesh(voxelGeo, new THREE.MeshBasicMaterial(
+      { color: 0xff0000, opacity: 0.4, transparent: true }));
+    scene.add(placementVisualizer);
+    placementVisualizer.add(new THREE.LineSegments(outlineGeometry, outlineMaterial))
+
+    let voxels = []
+    function Voxel() {
+      const voxel = new THREE.Mesh(voxelGeo, voxelMaterial);
+      voxel.add(new THREE.LineSegments(outlineGeometry, outlineMaterial))
+      voxels.push(voxel)
+      objectsOnWhichVoxelsCanBePlaced.push(voxel);
+      scene.add(voxel);
+      return voxel
+    }
+
+    {
+      var eraser = new THREE.Mesh(new THREE.PlaneBufferGeometry(.1,.1),new THREE.MeshBasicMaterial({transparent:true,color:0xFF0000}))
+      camera.add(eraser)
```

Comment:
The eraser should be an HTML button, rather than a THREE.js object (e.g. for accessibility)